engine/src/main/java/org/hibernate/search/engine/backend/document/model/dsl/IndexSchemaElement.java

```diff
@@ -7,19 +7,73 @@
 package org.hibernate.search.engine.backend.document.model.dsl;
 
 
+import java.util.function.Function;
+
+import org.hibernate.search.engine.backend.document.IndexFieldAccessor;
 import org.hibernate.search.engine.backend.types.dsl.IndexFieldTypeFactoryContext;
+import org.hibernate.search.engine.backend.types.dsl.IndexFieldTypeTerminalContext;
+import org.hibernate.search.engine.backend.types.IndexFieldType;
 
 /**
  * @author Yoann Rodiere
  */
 public interface IndexSchemaElement {
 
-	IndexFieldTypeFactoryContext field(String relativeFieldName);
+	/**
+	 * Add a field to this index schema element with the given type.
+	 *
+	 * @param relativeFieldName The relative name of the new field.
+	 * @param type The type of the new field.
+	 * @param <F> The type values help by the field.
```

Comment:
`The type values help by the field.` I don't understand this javadoc? `help` seems wrong but maybe I'm missing something? Same for all the other methods below.